Sources/OpenAPI/OpenApiProvider.swift

```diff
@@ -0,0 +1,184 @@
+import Vapor
+
+public final class OpenApiProvider: Provider {
+    public func register(_ s: inout Services) throws {
+        
+    }
+}
+
+public struct OpenAPIDefaults {
+    public let info: OpenAPIInfo
+
+}
+
+public struct OpenApiService {
+    public let info: OpenAPIInfo
+
+    var routes: [Route]
+    
+    func getOpenAPISpec() -> Codable {
+        fatalError()
+    }
+}
+
+public struct OpenAPIInfo: Content {
+    public let title: String
+    public let description: String?
+    public let version: String
+    
+    public let termsOfService: URL?
+    
+    public let contact: OpenAPIContact?
+    
+    public let license: OpenAPILicense?
+}
+
+public struct OpenAPIContact: Content {
```

Comment:
I think it could make sense to nest this struct in `OpenAPIInfo`. The same for any other structs that can be nested, too.

Reply:
Agreed, the reason I waited with that is to see if any structs are cross referenced, if not, I'll be nesting them :)